bubuku/features/rebalance/change_simple.py

```diff
@@ -0,0 +1,443 @@
+                _LOG.warning('Partition {} {} is lost, cause no broker have it'.format(topic, partition))
+                continue
+            p = Partition(topic, str(partition), [self.active_brokers.get(str(id_), self.fake) for id_ in broker_ids])
+            self.partitions[p.get_key()] = p
+
+    def createInitialDistribution(self):
+        total_partitions = 0
+        total_leaders = 0
+        az_counts = {}
+        for broker in self.active_brokers.values():
+            az_counts[broker.zone] = az_counts.get(broker.zone, 0) + 1
+        self.zone_checker = ZoneChecker([zone for zone in az_counts.keys()])
+
+        for partition in self.partitions.values():
+            replication_factor = len(partition.brokers)
+
+            improper_brokers = self.zone_checker.get_improper_brokers(partition.brokers)
+            for broker_idx in range(0, len(partition.brokers)):
+                added = False
+                broker = partition.brokers[broker_idx]
+                if broker.get_id() in self.active_brokers and broker not in improper_brokers:
+                    if broker.add_partition(broker_idx == 0, partition):
+                        added = True
+                if not added:
+                    partition.brokers[broker_idx] = self.fake
+                    self.fake.add_partition(broker_idx == 0, partition)
+                    self.register_partition_change(partition)
+
+            total_partitions += replication_factor
+            total_leaders += 1
+
+        leaders_min, leaders_max = get_division_result_min_max(total_leaders, len(self.active_brokers))
+
+        per_zone_partitions_min, per_zone_partitions_max = get_division_result_min_max(total_partitions, len(az_counts))
+
+        for zone, count in az_counts.items():
+            per_broker_partitions_min, _ = get_division_result_min_max(per_zone_partitions_min, count)
+            _, per_broker_partitions_max = get_division_result_min_max(per_zone_partitions_max, count)
+            for broker in self.active_brokers.values():
+                if broker.zone == zone:
+                    broker.set_partitions_estimation(
+                        (per_broker_partitions_min, per_broker_partitions_max),
+                        (leaders_min, leaders_max),
+                    )
+
+    def empty_fake(self):
+        leaders_load = sorted(self.active_brokers.values(), key=lambda x: x.get_load_factor(True))
+        replica_load = sorted(self.active_brokers.values(), key=lambda x: x.get_load_factor(False))
+
+        for pkey, is_leader in self.fake.list_partition_keys():
+            load = leaders_load if is_leader else replica_load
+            partition = self.partitions[pkey]
+            if not transfer_partition(partition, self.fake, load, self.zone_checker, polite=False):
+                raise Exception('Nowhere to transfer partition {}'.format(pkey))
+            self.register_partition_change(partition)
+            if is_leader:
```

Comment:
why do you need this `if` clause? you also pass `x.get_load_factor(True)` `True` in both cases